browser/ui/commands/accelerator_service_unittest.cc

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/ui/commands/accelerator_service.h"
+#include "base/test/scoped_feature_list.h"
+#include "brave/components/commands/browser/accelerator_pref_manager.h"
+#include "brave/components/commands/common/accelerator_parsing.h"
+#include "brave/components/commands/common/features.h"
+#include "chrome/app/chrome_command_ids.h"
+#include "chrome/test/base/testing_profile.h"
+#include "components/sync_preferences/testing_pref_service_syncable.h"
+#include "content/public/test/browser_task_environment.h"
+#include "testing/gtest/include/gtest/gtest.h"
+#include "third_party/googletest/src/googletest/include/gtest/gtest.h"
+
+class AcceleratorServiceUnitTest : public testing::Test {
+ public:
+  AcceleratorServiceUnitTest() {
+    features_.InitAndEnableFeature(commands::features::kBraveCommands);
+  }
+
+  ~AcceleratorServiceUnitTest() override = default;
+
+  void SetUp() override {
+    commands::AcceleratorPrefManager::RegisterProfilePrefs(
+        profile().GetTestingPrefService()->registry());
+  }
+
+  TestingProfile& profile() { return profile_; }
+
+ private:
+  content::BrowserTaskEnvironment task_environment_;
+  TestingProfile profile_;
+  base::test::ScopedFeatureList features_;
+};
+
+TEST_F(AcceleratorServiceUnitTest, CanOverrideExistingShortcut) {
+  commands::AcceleratorService service(
+      profile().GetPrefs(),
+      {{IDC_NEW_TAB, {commands::FromCodesString("Control+KeyT")}}});
+
+  auto accelerators = service.GetAcceleratorsForTesting();
+  ASSERT_EQ(1u, accelerators[IDC_NEW_TAB].size());
+  EXPECT_EQ("Control+KeyT",
+            commands::ToCodesString(accelerators[IDC_NEW_TAB][0]));
+
+  service.AssignAcceleratorToCommand(IDC_NEW_WINDOW, "Control+KeyT");
+  accelerators = service.GetAcceleratorsForTesting();
+  EXPECT_EQ(0u, accelerators[IDC_NEW_TAB].size());
+  ASSERT_EQ(1u, accelerators[IDC_NEW_WINDOW].size());
```

Comment:
Yeah, it's an `ASSERT_EQ` because I index into the array on the next line, so I'd prefer it crash here than potentially do something random when I access the first item